src/vaadin-menu-bar-submenu.html

```diff
@@ -28,6 +28,19 @@
       _openedChanged(opened) {
         this.$.overlay.opened = opened;
       }
+
+
+      /**
+       * Overriding the public method to reset expanded button state.
+       */
+      close() {
+        super.close();
+
+        const host = this.getRootNode().host;
+        if (host && host.tagName.toLowerCase() === 'vaadin-menu-bar') {
```

Comment:
Hmm...this wouldn't work with menu-bar extensions because of the fixed tag name, right?

Reply:
Thanks, fixed. 
Now checking if submenu is not under `document` (which means it's not a nested one).